velox/exec/HashBuild.cpp

```diff
@@ -794,19 +842,35 @@ void HashBuild::ensureTableFits(uint64_t numRows) {
     return;
   }
 
+  if (!allowDuplicateRows) {
```

Comment:
Shall we move test injection before duplicate row reservation?

Reply:
test injection is done right after we get total number of rows. If total number of rows equals to 0 then we don't want to trigger spill as there is nothing to spill, which is the original logic.
We have 3 ways:
1) keep it at the current location
2) change the logic, so that even when total num rows is 0 we still trigger test spill.
3) I can extract the total num rows fetching and memory reservation out separately. But this will make the code look not so clean.

Reply:
We should do test spill injection before any reserve memory calls.